cubedash/_stac.py

```diff
@@ -266,7 +266,31 @@ def as_stac_item(dataset: DatasetItem) -> pystac.Item:
     return item
 
 
+def _provider_from_dict(d: dict) -> pystac.Provider:
+    """A custom parser that extends pystac.Provider.from_dict()"""
+    # pystac.Provider.from_dict() does not validate roles against ProviderRole
+    # or ensure roles are in a list (in case we provide a string instead of a list)
+    p = pystac.Provider.from_dict(d)
+    roles = p.roles  # unvalidated
+    if roles is not None:
+        p.roles = []  # reset
+        if isinstance(roles, str):
+            roles = [roles]
+        for r in roles:
+            if r is None:
```

Comment:
I don't think null values should be quietly ignored; either we should add a warning here or let it be handled like any other invalid provider role.